Classes/TextUtils.cs

```diff
@@ -24,19 +24,106 @@ namespace RadioDld
 
     public static class TextUtils
     {
+        // Move Static RegEx content here to declutte the 'StripDateFromName' function
+        // Declarations
+
+        // Key date formats to look out for
+        // 1 / d(dd)(st | nd | rd | th) | m(mm)(mmm) | yyyy('yy)(yy)
+        // 2 / yyyy('yy)(yy)|m(mm)(mmm)|d(dd)(st|nd|rd|th)
+        // 3 / mmm|d(dd)(st|nd|rd|th)|yyyy('yy)(yy)
+
+        // ## Common Delims
+        //    .| -|/|{ ws}
+        //
+        // ###Test Beds
+        // https://www.debuggex.com/
+        // http://regexstorm.net/tester
+        // https://regex101.com/
+        //
+        // new RegEx in Java
+        // (((3[01]|2\d|1\d|0?\d)(st|nd|rd|th)?(\.|\-|\/| )((2\d|1\d|0?\d)|(?:Jan(?:uary)?|Feb(?:ruary)?|Mar(?:ch)?|Apr(?:il)?|May|Jun(?:e)?|Jul(?:y)?|Aug(?:ust)?|Sep(?:t(?:ember)?)?|Oct(?:ober)?|Nov(?:ember)?|Dec(?:ember)?))(\.|\-|\/| )(\d{4}|\'\d{2}|\d{2}))( ?)|((\d{4}|\'\d{2}|\d{2})(\.|\-|\/| )((2\d|1\d|0?\d)|(?:Jan(?:uary)?|Feb(?:ruary)?|Mar(?:ch)?|Apr(?:il)?|May|Jun(?:e)?|Jul(?:y)?|Aug(?:ust)?|Sep(?:t(?:ember)?)?|Oct(?:ober)?|Nov(?:ember)?|Dec(?:ember)?))(\.|\-|\/| )(3[01]|2\d|1\d|0?\d)(st|nd|rd|th)?( ?))|((?:Jan(?:uary)?|Feb(?:ruary)?|Mar(?:ch)?|Apr(?:il)?|May|Jun(?:e)?|Jul(?:y)?|Aug(?:ust)?|Sep(?:t(?:ember)?)?|Oct(?:ober)?|Nov(?:ember)?|Dec(?:ember)?))(\.|\-|\/| )(3[01]|2\d|1\d|0?\d)(st|nd|rd|th)?(\.|\-|\/| )((?:(?:\d{4})|(?:\d{2})(?![\\d])|(?:\')(?:\d{2})(?![\\d])))( ?))
+
+        // Common Groups
+        private static string reDelim = @"(\.|\-|\/| )"; // Common Delims .| -|/|{ ws}
```

Comment:
Few things I'm afraid:
- Please don't use hungarian notation (eg. the variable name prefixes relating to the type)
- The strings used to build the regular expressions should be const
- I don't think that the list of what is matched by the regex should be included after the description of the regex in the comments, as this is already included in the regex itself.

Reply:
Can you help me understand why using Hungarian notion isn't okay? (had to look up what Hungarian was). And perhaps give me an example of what notion format you would use instead for those variables? I'm thinking (from looking at some of your other code) that ReDelim or REDelim might okay?

okay Static to Const that should be easy enough :)

> I don't think that the list of what is matched by the regex should be included after the description of the regex in the comments, as this is already included in the regex itself.

Not sure which lines of code you are referring to exactly... i know i've got a lot of comments in code, but that my way of being clear on what the code is (suppose) to do. I'll remove a significant amount when closer to a final version but maybe not too much with this RegEx given it so complex might be easier for the next person trying to fix something i missed (or messed up).

Reply:
> Can you help me understand why using Hungarian notion isn't okay? (had to look up what Hungarian was). And perhaps give me an example of what notion format you would use instead for those variables? I'm thinking (from looking at some of your other code) that ReDelim or REDelim might okay?

https://softwareengineering.stackexchange.com/a/102690 makes some good points, but essentially in this case, it needs to match the rest of the codebase for consistency.

I'd suggest something like `MatchDelim` or `DelimPattern` would be reasonably descriptive. The aim is to describe what the variable is or does (not that I'm always successful).

> Not sure which lines of code you are referring to exactly... i know i've got a lot of comments in code, but that my way of being clear on what the code is (suppose) to do. I'll remove a significant amount when closer to a final version but maybe not too much with this RegEx given it so complex might be easier for the next person trying to fix something i missed (or messed up).

Hopefully the below makes it a bit clearer - the part I've highlighted duplicates the regex but in a non-standard format.

```
private static string reDelim = @"(\.|\-|\/| )"; // Common Delims .| -|/|{ ws}
                                                                  ^^^^^^^^^^^^ 
```
The main issue with this is that when the code changes, comments are often not updated which is confusing to programmers making changes in future.